src/game/shared/tf/tf_weapon_shotgun.cpp

```diff
@@ -329,27 +328,29 @@ void CTFScatterGun::FireBullet( CTFPlayer *pPlayer )
 			return;
 
 		// Knock the firer back!
-		if ( !(pOwner->GetFlags() & FL_ONGROUND) && !pPlayer->m_bScattergunJump )
+		if ( !(pOwner->GetFlags() & FL_ONGROUND) && !pPlayer->m_Shared.m_bScattergunJump )
 		{
-			pPlayer->m_bScattergunJump = true;
+			pPlayer->m_Shared.m_bScattergunJump = true;
 
+#ifndef CLIENT_DLL
 			pOwner->m_Shared.StunPlayer( 0.3f, 1.f, TF_STUN_MOVEMENT | TF_STUN_MOVEMENT_FORWARD_ONLY );
+#endif
 
 			float flForce = AirBurstDamageForce( pOwner->WorldAlignSize(), 60, 6.f );
 
 			Vector vecForward;
 			AngleVectors( pOwner->EyeAngles(), &vecForward );
 			Vector vecForce = vecForward * -flForce;
 
-			EntityMatrix mtxPlayer;
```

Comment:
Maybe we should just bring EntityMatrix over for client as well?

Reply:
sounds fair, i was trying to minimise how many files i touch for this fix

something to note about EntityMatrix::InitFromEntity is that it uses GetAbsAngles but the client should use EyeAngles otherwise there'd be pred errors when you FaN jump while moving your mouse